src/getCalculatedMeasures.ts

```diff
@@ -0,0 +1,57 @@
+import { ContentRecord } from "@activeviam/activeui-sdk";
+import xml2js from "xml2js";
+import { LegacyCalculatedMeasure } from "./migrateCalculatedMeasure";
+
+/**
+ * Extracts and parses the XML calculated measure objects from the /pivot/entitlements/cm folder.
+ */
+export const getCalculatedMeasures = async (
+  calculatedMeasuresFolder: ContentRecord
+): Promise<LegacyCalculatedMeasure[]> => {
+  const parser = new xml2js.Parser();
+
+  const calculatedMeasures: LegacyCalculatedMeasure[] = [];
+
+  if (calculatedMeasuresFolder?.children) {
+    await Promise.all(
+      Object.values(calculatedMeasuresFolder?.children).map(
+        async ({ children = {} }) => {
```

Comment:
I'm still confused by this: why are there two layers of children ?
It seems to me that the children of the calculated measures folder are the calculated measures files directly.

Reply:
There's two levels of `children` here because the shape of the `calculatedMeasuresFolder` is as follows: 

```
{
    "entry": {...},
     "children": {
        "EquityDerivativesCube": {
           "entry": {...},
            "children": {
               // The actual measure objects live here
           }
     }
}
```

As you can see above, the first level of `children` grabs the `EquityDerivativesCube` object, but since in AUI-5, we don't distinguish the measures based on cube, I grab just the `values` with `Object.values`. I then grab the second level of `children` which actually contains the measures.

Reply:
Ah yes I forgot about the cubes.
Can you add a comment in the code ?
Something like:
"Legacy calculated measures are grouped by cube in `/pivot/entitlements/cm`. But they are flattened in ActiveUI 5".
And then you could use `_flatMap`.
Lastly, instead of defaulting `children` to `{}`, I would handle this case with an early return: it's more explicit so easier to read, and there is no unnecessary extra work.